Lib/test/test_syslog.py

```diff
@@ -78,6 +79,62 @@ def logger():
         finally:
             sys.setswitchinterval(orig_si)
 
+    def test_subinterpreter_syslog(self):
+        # syslog.syslog() is not allowed in subinterpreters, but only if
+        # syslog.openlog() hasn't been called in the main interpreter yet.
+        with self.subTest('before openlog()'):
+            code = dedent('''
+                import syslog
+                caught_error = False
+                try:
+                    syslog.syslog('foo')
+                except RuntimeError:
+                    caught_error = True
+                assert(caught_error)
+            ''')
+            res = support.run_in_subinterp(code)
+            self.assertEqual(res, 0)
+
+        syslog.openlog()
+        with self.subTest('after openlog()'):
+            code = dedent('''
+                import syslog
+                syslog.syslog('foo')
+            ''')
+            res = support.run_in_subinterp(code)
+            self.assertEqual(res, 0)
+        syslog.closelog()
+
+    def test_subinterpreter_openlog(self):
+        code = dedent('''
+            import syslog
+            caught_error = False
+            try:
+                syslog.openlog()
+            except RuntimeError:
+                caught_error = True
+
+            assert(caught_error)
+        ''')
+        res = support.run_in_subinterp(code)
+        self.assertEqual(res, 0)
```

Comment:
Well, when considering follow implementation
https://github.com/python/cpython/blob/e37744f289af00c6f6eba83f7abfb932b63de9e0/Lib/test/test_syslog.py#L16

It might be not needed but I like your suggestion :)

Reply:
So please let me know if we have to revert to your suggestion even if we consider the tearDown method.
If not I will merge this PR by this weekend.

Reply:
My only concern was that `syslog.closelog()` wasn't called sometimes.  It's not important to me how we fix that. 😄